tests/ert_tests/data/test_measured_data.py

```diff
@@ -110,42 +110,112 @@ def test_remove_failed_realizations(
+                data=[[1.0, 3.0], [4.0, 6.0], [7.0, 9.0]],
+                index=["OBS", "STD", 1],
+                columns=[0, 2],
+            ),
+            None,
+            id="none-in-STD-row",
+        ),
+        pytest.param(
+            pd.DataFrame(data=[[1, 2, 3], [4, 5, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[7.0, np.nan, 9.0]], index=[1]),
+            pd.DataFrame(
+                data=[[1.0, 3.0], [4.0, 6.0], [7.0, 9.0]],
+                index=["OBS", "STD", 1],
+                columns=[0, 2],
+            ),
+            None,
+            id="nan-in-measured-data",
+        ),
+        pytest.param(
+            pd.DataFrame(data=[[1, 2, 3], [4, 5, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[7.0, np.inf, 9.0]], index=[1]),
             pd.DataFrame(
                 data=[[1.0, 3.0], [4.0, 6.0], [7.0, 9.0]],
                 index=["OBS", "STD", 1],
                 columns=[0, 2],
             ),
+            None,
+            id="inf-in-measured-data",
+        ),
+        pytest.param(
+            pd.DataFrame(data=[[1, 2, 3], [4, 5, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[7.0, -np.inf, 9.0]], index=[1]),
+            pd.DataFrame(
+                data=[[1.0, 3.0], [4.0, 6.0], [7.0, 9.0]],
+                index=["OBS", "STD", 1],
+                columns=[0, 2],
+            ),
+            None,
+            id="negative-inf-in-measured-data",
+        ),
+        pytest.param(
+            pd.DataFrame(data=[[1, 2, 3], [4, 5, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[np.inf, np.inf, np.inf]], index=[1]),
+            pd.DataFrame(data=[], index=["OBS", "STD", 1]),
+            (ValueError, "This operation results in an empty dataset"),
+            id="all-inf-in-measured-data",
+        ),
+        pytest.param(
+            pd.DataFrame(data=[[1, 2, 3], [4, 5, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[np.nan, np.nan, np.nan]], index=[1]),
+            pd.DataFrame(data=[], index=["OBS", "STD", 1]),
+            (ValueError, "This operation results in an empty dataset"),
+            id="all-nan-in-measured-data",
         ),
     ],
 )
 def test_remove_inactive_observations(
-    input_dataframe,
+    input_header,
+    measured_data,
     expected_result,
+    expected_error,
     monkeypatch,
     facade,
     measured_data_setup,
-    valid_data,
 ):
-    input_dataframe.columns = _set_multiindex(input_dataframe)
-    measured_data_setup(valid_data, input_dataframe, monkeypatch)
+    input_header.columns = _set_multiindex(input_header)
+    measured_data_setup(measured_data, input_header, monkeypatch)
     md = MeasuredData(facade, ["obs_key"])
 
     expected_result.columns = _set_multiindex(expected_result)
     expected_result = pd.concat({"obs_key": expected_result}, axis=1)
 
-    md.remove_inactive_observations()
-    assert md.data.equals(expected_result)
+    if expected_error is None:
```

Comment:
Dont think we should fork the test here, it would be better to split it into two different tests, one for valid input, and one for invalid input.

Reply:
done.